altlas/Assets/scripts/MapGenerator.cs

```diff
@@ -290,11 +291,65 @@ public void spawnStacksInRow(float horLength, float verLength, List<List<MapData
         }
     }
 
-    /**
-     * spawnStackInRow with fixed drawer length
-     * */
-    public void spawnRowInDrawer(MapData[][] stacks, string category) {
-        spawnStacksInRow(DRAWER_HOR_LEN, DRAWER_VER_LEN, stacks, category);
+    void setLabelOnDraw(KeyValuePair<string, Dictionary<string, List<MapData>>> category)
+    {     
+        GameObject go = new GameObject("myText");
+        go.AddComponent<TextMesh>();
+        TextMesh tm = go.GetComponent(typeof(TextMesh)) as TextMesh;
+        go.name = category.Key + "_label";
```

Comment:
the value that you're passing to the GameObject constructor is the name - thus you could remove this line and pass category.key + label to the constructor instead